server/routes/commonPostApi.js

```diff
@@ -0,0 +1,18 @@
+const express = require('express');
+
+const router = express.Router();
+
+const postController = require('../controllers/userPostController');
+const postValidator = require('../validators/postValidator');
+
+router.get('/posts', postValidator.getAllPostValidation, postController.getAllPost);
+router.get('/posts/:id', postValidator.getPostValidation, postController.getPost);
+router.post('/posts/:id/addStudent', postValidator.addStudentValidation, postController.addStudentToPost);
+router.get('/:id/applications', postValidator.applicationsValidation, postController.getAllApplications);
+router.post('/:id/addFavourite', postValidator.addFavouriteValidation, postController.addFavouritePostToStudent);
```

Comment:
@tauoverpi  @devanshu0987  are we keeping a common post api for sure?? there will be some situations in which the  way the posts are viewed will differ [eg : admin side favourites ; apply will not be there ; there will be some posts which will be displayed only on the admin side ] - do we deal with these on the client side or have a different api for admin and user [ common model ]

Reply:
Right now we have assigned all the rights of an user to admin  i.e. an admin can do whatever an user can + some more. I know this might not be the best thing, but @in42 and I had a bit of discussion and we decided to go along with that, as that felt best at that moment,